.github/workflows/git-builds.yaml

```diff
@@ -7,38 +7,24 @@ on:
     - cron: '17 08,14 * * 1-5'
   workflow_dispatch:
 
-jobs:
-  build:
-    name: Build (${{ matrix.os }})
-    runs-on: ${{ matrix.os }}
-
-    strategy:
-      matrix:
-        os: [ ubuntu-latest, windows-latest ]
-
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: "recursive"
-
-      - name: Info
-        run: dotnet --info
+env:
+  ProjectFile: "src/NexusMods.App/NexusMods.App.csproj"
 
-      - name: Build (Linux)
-        if: runner.os == 'Linux'
-        shell: sh
-        run: ./scripts/build-linux-x64.sh
-
-      - name: Build (Windows)
-        if: runner.os == 'Windows'
-        shell: pwsh
-        run: scripts\build-win-x64.ps1
-
-      - name: Upload artifact
-        uses: actions/upload-artifact@v3
-        with:
-          # example: "NexusMods.App-Windows-279a5092173a6f687f89dc3c6a2b7c5898e289f6"
-          name: NexusMods.App-${{ runner.os }}-${{ github.sha }}
-          path: ${{ github.workspace }}/bin
-          if-no-files-found: error
-          retention-days: 7
+jobs:
+  build-linux:
+    uses: ./.github/workflows/build-linux-pupnet.yaml
+    with:
+      AppVersion: "0.0.1"
```

Comment:
Does this get replaced later on or is this going to be the final version?

Reply:
It should be what it shows as, we use this to set the version during compilation (or rather, PupNet does).
And the App itself gets the version from its assembly, thus this propagates.

I was just considering the issue to be of a different kind than what is being solved here, so I figured I'd make a ticket for improving this instead and be fine with it for the time being.

Reply:
Namely we would want to discuss what format we want first. So I was considering raising that during standup.

I would prefer something like commit ID (`${{ github.sha }}`) but without testing it, I'm not sure if the current build script would accept that. It might expect a semver, so something like `0.0.0-${{ github.sha }}`.

Reply:
In MO2 we used Appveyor before, transitioning to GitHub actions now. 
There we would use the current app version and add a monotonically increasing number to it.

I would like to get the current app version from somewhere and append the sha of the commit.

Reply:
The artifact names already use the SHA:

![2024-06-03_12-18-09](https://github.com/Nexus-Mods/NexusMods.App/assets/16577545/a3b20e10-95b0-4929-8dda-d30dc1ca4d91)

Reply:
What would be gain by putting the SHA into the assembly version?

Reply:
The main issue is distinguishing versions from user logs. E.g. someone tests one of these in the office, and a few days later we don't remember to which build those logs referred to.

Reply:
It's mostly about being to determine which build we're looking at if someone, like @LukeNexusMods or @Pickysaurus  takes a screenshot.

Reply:
We show the SHA in the bottom banner of the UI, and it should also be in the logs, no?

Reply:
I guess we already have that information then. 
Do we still want to have these output an actual version rather than 0.0.1?

Reply:
We don't have an actual version. This CI runs twice a day, every work day. These builds are between versions.

Reply:
We use [sourcelink](https://github.com/dotnet/sourcelink/tree/main) that automatically injects the SHA into `AssemblyInformationalVersionAttribute`.

Reply:
I guess it be beneficial to know between which versions though. 
Maybe we can revisit this later if we see it is an issue.